runtime/compiler/ilgen/Walker.cpp

```diff
@@ -229,7 +229,11 @@ TR::Block * TR_J9ByteCodeIlGenerator::walker(TR::Block * prevBlock)
       //
       // NOTE: first request occurs in the switch statement over bytecodes,
       // second request occurs in stashArgumentsForOSR
-      static_cast<TR_ResolvedJ9JITServerMethod *>(comp()->getMethodBeingCompiled())->cacheResolvedMethodsCallees(2);
+      static_cast<TR_ResolvedJ9JITServerMethod *>(_methodSymbol->getResolvedMethod())->cacheResolvedMethodsCallees(2);
```

Comment:
The test on `comp()->getMethodBeingCompiled()` should be replaced with `_methodSymbol->getResolvedMethod()`.

Indeed, the walker is called for inlined methods as well. However, removing the caching from `estimateCallSize` increases the number of messages, which leads me to believe that code size estimation may happen before IL for inlined methods is generated. I should test if caching in walker just for the method being compiled does not result in message increase.

Reply:
I measured that caching resolved methods only for method being compiled in walker does not lead to any significant difference in the number of messages compared to caching for inlined methods as well. This makes sense, since if the methods were already cached during code size estimation, no remote call would be made from walker. We still waste some time iterating over bytecodes though. Thus, I think it's better to cache only for method being compiled.

Reply:
I assume `_methodSymbol->getResolvedMethod() == comp()->getMethodBeingCompiled()` would be a sufficient check.

Reply:
yes, that check is sufficient